demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -0,0 +1,841 @@
+                    if os.path.basename(rest_of_path) == PACKS_DIR:
+                        click.secho(f'\n================= Validating pack {path} =================',
+                                    fg="bright_cyan")
+                        valid_files.add(self.run_validations_on_pack(path))
+
+                    else:
+                        click.secho(f'\n================= Validating package {path} =================',
+                                    fg="bright_cyan")
+                        valid_files.add(self.run_validation_on_package(path, error_ignore_list))
+
+        return all(valid_files)
+
+    def run_validation_on_all_packs(self):
+        """Runs validations on all files in all packs in repo (a)
+
+        Returns:
+            bool. true if all files are valid, false otherwise.
+        """
+        click.secho('\n================= Validating all files =================', fg="bright_cyan")
+        valid_repo = set()
+
+        if not self.skip_conf_json:
+            print('\nValidating conf.json')
+            conf_json_validator = ConfJsonValidator()
+            valid_repo.add(conf_json_validator.is_valid_conf_json())
+
+        for pack_name in os.listdir(PACKS_DIR):
+            pack_path = os.path.join(PACKS_DIR, pack_name)
+            valid_repo.add(self.run_validations_on_pack(pack_path))
+
+        return all(valid_repo)
+
+    def run_validations_on_pack(self, pack_path):
+        """Runs validation on all files in given pack. (i,g,a)
+
+        Args:
+            pack_path: the path to the pack.
+
+        Returns:
+            bool. true if all files in pack are valid, false otherwise.
+        """
+        valid_pack = set()
+        pack_error_ignore_list = self.get_error_ignore_list(os.path.basename(pack_path))
+
+        valid_pack.add(self.validate_pack_unique_files(pack_path, pack_error_ignore_list))
+
+        for content_dir in os.listdir(pack_path):
+            content_entity_path = os.path.join(pack_path, content_dir)
+            if content_dir in CONTENT_ENTITIES_DIRS:
+                valid_pack.add(self.run_validation_on_content_entities(content_entity_path, pack_error_ignore_list))
+
+        return all(valid_pack)
+
+    def run_validation_on_content_entities(self, content_entity_dir_path, pack_error_ignore_list):
+        """Gets non-pack folder and runs validation within it (Scripts, Integrations...)
+
+        Returns:
+            bool. true if all files in directory are valid, false otherwise.
+        """
+        valid_directory = set()
+        for file_name in os.listdir(content_entity_dir_path):
+            file_path = os.path.join(content_entity_dir_path, file_name)
+            if os.path.isfile(file_path):
+                if file_path.endswith('.json') or file_path.endswith('.yml') or file_path.endswith('.md'):
+                    valid_directory.add(self.run_validations_on_file(file_path, pack_error_ignore_list))
+
```

Comment:
maybe add ignored files here?

Reply:
not sure which files do you mean here

Reply:
.py files
or pack_ignore in the upper funciton